core/src/layers/prometheus.rs

```diff
@@ -147,77 +224,68 @@ impl PrometheusLayer {
         self.path_label_level = level;
         self
     }
-}
 
-impl<A: Access> Layer<A> for PrometheusLayer {
-    type LayeredAccess = observe::MetricsAccessor<A, PrometheusInterceptor>;
-
-    fn layer(&self, inner: A) -> Self::LayeredAccess {
-        let interceptor = PrometheusInterceptor::register(
-            self.registry.clone(),
-            self.operation_duration_seconds_buckets.clone(),
-            self.operation_bytes_buckets.clone(),
-            self.path_label_level,
-        );
-        observe::MetricsLayer::new(interceptor).layer(inner)
-    }
-}
-
-#[derive(Clone, Debug)]
-pub struct PrometheusInterceptor {
-    operation_duration_seconds: HistogramVec,
-    operation_bytes: HistogramVec,
-    operation_errors_total: GenericCounterVec<AtomicU64>,
-    path_label_level: usize,
-}
-
-impl PrometheusInterceptor {
-    fn register(
-        registry: Registry,
-        operation_duration_seconds_buckets: Vec<f64>,
-        operation_bytes_buckets: Vec<f64>,
-        path_label_level: usize,
-    ) -> Self {
-        let labels = OperationLabels::names(false, path_label_level);
-        let operation_duration_seconds = register_histogram_vec_with_registry!(
+    /// Register the metrics into the given registry and return a [`PrometheusLayer`].
+    pub fn register(self, registry: &Registry) -> PrometheusLayer {
```

Comment:
How about returning an error instead of just unwrapping?

Reply:
Should we return `prometheus::Error` or `opendal::Error`?

Reply:
opendal only returns `opendal::Error`, we can wrap it inside.